kview/src/main/java/dev/ikm/komet/kview/klfields/KlFieldHelper.java

```diff
@@ -108,17 +116,62 @@ public static List<ObservableField<?>> generateObservableFieldsAndNodes(ViewProp
 
         return observableFields;
     }
 
+    public static int generateHashValue(Latest<SemanticEntityVersion> semanticEntityVersionLatest, ViewProperties viewProperties ) {
+        List<ObservableField<?>> observableFieldsList = new ArrayList<>();
+        Consumer<FieldRecord<Object>> fieldRecordConsumer = (fieldRecord) -> {
+            ObservableField<?> writeObservableField = obtainObservableField(viewProperties, semanticEntityVersionLatest, fieldRecord, false);
+            ObservableField<?> observableField = new ObservableField<>(writeObservableField.field(), false);
+            observableFieldsList.add(observableField);
+        };
+        generateSemanticUIFields(viewProperties, semanticEntityVersionLatest, fieldRecordConsumer);
+        return calculteHashValue(observableFieldsList);
+    }
+
+    /**
+     * This method will return the latest commited version.
+     * //TODO this method can be generalized to return latest<EntityVersion> As of now it is just returning SemanticEntityVersion.
+     * // TODO need to implement logic for create Semantic.
+     *
+     * @return entityVersionLatest
+     * */
+    public static Latest<SemanticEntityVersion> retrieveCommittedLatestVersion(EntityFacade entityFacade, ViewProperties viewProperties) {
+        AtomicReference<Latest<SemanticEntityVersion>> entityVersionLatest = new AtomicReference<>();
+
+        StampCalculator stampCalculator = viewProperties.calculator().stampCalculator();
+        //retrieve latest semanticVersion
+        Latest<SemanticEntityVersion> semanticEntityVersionLatest = stampCalculator.latest(entityFacade.nid());
+        ObservableSemantic observableSemantic = ObservableEntity.get(semanticEntityVersionLatest.get().nid());
+        ObservableSemanticSnapshot observableSemanticSnapshot = observableSemantic.getSnapshot(viewProperties.calculator());
+        //Get list of previously committed data sorted in latest at the top.
+        ImmutableList<ObservableSemanticVersion> observableSemanticVersionImmutableList = observableSemanticSnapshot.getHistoricVersions();
+        // Filter out Uncommitted data. Data whose time stamp parameter is Long.MAX_VALUE. and get the 1st available.
+        Optional<ObservableSemanticVersion> observableSemanticVersionOptional = observableSemanticVersionImmutableList.stream().filter(p -> p.stamp().time() != Long.MAX_VALUE).findFirst();
+
+        observableSemanticVersionOptional.ifPresentOrElse( (p) -> {
+            entityVersionLatest.set(new Latest<>(p));
+        }, () -> {entityVersionLatest.set(semanticEntityVersionLatest);});
+
+
+        return entityVersionLatest.get();
+    }
+
+    public static int calculteHashValue(List<ObservableField<?>> observableFieldsList ) {
+        StringBuilder stringBuilder = new StringBuilder();
+        observableFieldsList.forEach(observableField -> {
+            stringBuilder.append(observableField.valueProperty().get().toString()).append("|");
```

Comment:
Maybe out of scope for this issue but we might want to be careful and not just use toString for complex values like IntIdSet and IntIdList but actually iterate through those collections (when that's the case) and do toString on each item. toString for those collections are usually really meant for debugging purposes, etc and not as a reliable source of how whether the list or set was changed. Example: a toString for a Set or List might return the same value for 2 different Sets or Lists when the only difference is the order that they have. 
Bottomline toString for a complex object (IntIdSet or IntIdList) is not reliable as a way to know exactly what they contain. But again might be out of scope for this PR.